dev-client/src/components/map/SiteMap.tsx

```diff
@@ -1,23 +1,76 @@
-import Mapbox, {Camera, MarkerView, UserLocation} from '@rnmapbox/maps';
-import React, {memo, useEffect, useRef, useState} from 'react';
+import Mapbox, {Camera, UserLocation} from '@rnmapbox/maps';
+import {OnPressEvent} from '@rnmapbox/maps/src/types/OnPressEvent';
+import {memo, useEffect, useMemo, useRef, useState} from 'react';
 // TODO: Is it better to import type?
 import {type Position} from '@rnmapbox/maps/lib/typescript/types/Position';
 import MaterialIconButton from '../common/MaterialIconButton';
 import {v4 as uuidv4} from 'uuid';
-
-export type SiteMarker = {
-  id: string;
-  coordinates: GeoJSON.Position;
-  popup?: JSX.Element;
-};
+import {useSelector} from '../../model/store';
+import {Site} from 'terraso-client-shared/site/siteSlice';
+import {Box, Heading, Text} from 'native-base';
 
 type SiteMapProps = {
-  sites: SiteMarker[];
   center?: Position;
 };
 
-const SiteMap = memo(({sites, center}: SiteMapProps): JSX.Element => {
-  const [temporarySites, setTemporarySites] = useState<SiteMarker[]>([]);
+const siteFeatureCollection = (
+  sites: Site[],
+): GeoJSON.FeatureCollection<GeoJSON.Geometry> => ({
+  type: 'FeatureCollection',
+  features: sites.map(site => ({
+    type: 'Feature',
+    id: site.id,
+    properties: {},
+    geometry: {
+      type: 'Point',
+      coordinates: [site.longitude, site.latitude],
+    },
+  })),
+});
+
+const SiteCallout = ({site}: {site: Site}): JSX.Element => {
+  return (
+    <Mapbox.MarkerView
+      coordinate={[site.longitude, site.latitude]}
+      anchor={{x: 0.5, y: 0}}
+      allowOverlap={true}>
+      <Box backgroundColor="background.default">
```

Comment:
```suggestion
      <Box bgColor="background.default">
```

Seems to be what the [style docs](https://docs.nativebase.io/next/utility-props#h3-color-and-background-color) say.